### PR TITLE
Support prefix-list BGP policies

### PR DESCRIPTION
**Pull request by wenovus**
* Convert OC policies to GoBGP policies in configuration in the reconciler.
* Add BGP policy test spec. This is adapted from an existing one.
* Add prefix-list policy test.

### CODE REVIEW

File: bgp/tests/proto/policyval/generate.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2023 Google LLC
```
Comment (DanG100) [inline bgp/tests/proto/policyval/generate.go:1]: shouldn't need this or update.sh. just use genproto.sh

File: bgp/tests/proto/policyval/policy_validation.proto [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package policy_validation;
```
Comment (DanG100) [inline bgp/tests/proto/policyval/policy_validation.proto:17]: is reason with a proto and not just a struct?

File: bgp/tests/local_tests/ygnmi_test.go [hunk 3]
```diff
@@ -93,9 +101,81 @@ func Delete[T any](t testing.TB, c *ygnmi.Client, q ygnmi.ConfigQuery[T]) *ygnmi
...
+	var st interface {
+		GRPCStatus() *status.Status
+	}
+	ok := errors.As(err, &st)
+	return ok && (st.GRPCStatus().Code() == codes.DeadlineExceeded || st.GRPCStatus().Code() == codes.Canceled)
+}
+
+// statusErr is an interface implemented by errors returned by gRPC.
+// https://pkg.go.dev/google.golang.org/grpc@v1.48.0/internal/status#Error
+type statusErr interface {
```
Comment (DanG100) [inline bgp/tests/local_tests/ygnmi_test.go:137]: fix lint issue

File: bgp/tests/proto/policyval/policy_validation.proto (hunk 2 above)
Comment (wenovus) [reply bgp/tests/proto/policyval/policy_validation.proto:17]: So there is another BGP policy testing framework that the user of this is looking at that currently uses this as the test interface. I'm simply using it here for uniformity and so I don't need to re-invent this. If there is no need for using a proto we can later change it, but it doesn't seem to have any harm.

File: bgp/tests/proto/policyval/update.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (DanG100) [inline bgp/tests/proto/policyval/update.sh:1]: can remove this too, covered by genproto.sh

**Review by DanG100 — APPROVED** [review]

**Comment by wenovus** [comment]
There are some strange non-deterministic errors appearing:

Locally running `make test-race`, this happened once:
```
--- FAIL: TestBGPGUEPolicy (20.45s)                                                                                                                                                                                                                                                                   
    --- FAIL: TestBGPGUEPolicy/v6 (10.20s)                                                                                                                                                                                                                                                            
        --- FAIL: TestBGPGUEPolicy/v6/Add_Policy (0.00s)                                                                                                                                                                                                                                              
            server_test.go:1950: Got duplicate route key:                                                                                          
                First: &{RouteKey:{Prefix:2001::1400:0:0:0/72 NIName:DEFAULT} Nexthops:map[{NextHopSummary:{Weight:0 Address:2001::c0a8:12a:0:0 NetworkInstance:DEFAULT} Port:{Name:eth0 Index:0 Subinterface:0} GUEHeaders:{GUEPolicy:{dstPortv4:0 dstPortv6:8 srcIP4:[0 0 0 0] srcIP6:[32 1 0 0 0 0 
0 0 42 42 42 42 0 0 0 0]} dstIP4:[0 0 0 0] dstIP6:[32 1 0 0 0 0 0 0 192 168 1 42 0 0 0 0] isV6:true}}:true]}                                                                                                                                                                                          
                Duplicate: &{RouteKey:{Prefix:2001::1400:0:0:0/72 NIName:DEFAULT} Nexthops:map[{NextHopSummary:{Weight:0 Address:2001::c0a8:12a:0:0 NetworkInstance:DEFAULT} Port:{Name:eth0 Index:0 Subinterface:0} GUEHeaders:{GUEPolicy:{dstPortv4:0 dstPortv6:8 srcIP4:[0 0 0 0] srcIP6:[32 1 0 0 
0 0 0 0 42 42 42 42 0 0 0 0]} dstIP4:[0 0 0 0] dstIP6:[32 1 0 0 0 0 0 0 192 168 1 42 0 0 0 0] isV6:true}}:true]}
```

With GitHub actions I've commented where bgp/tests/local_tests is failing due to a stale cache update -- I have no idea how that could happen since all gNMI operations for these tests happen sequentially, and tests are also run sequentially since they're in the same package: https://github.com/openconfig/lemming/actions/runs/5050247039/jobs/9060715556

Lastly there was a recursive cmp transformer failure that I also have no idea happened: https://github.com/openconfig/lemming/actions/runs/5050317179/jobs/9060884938

This is just a note for the future if these errors occur to help with debugging.

**Pull request merged**